coralproject/talk
# Ability to extend Configuration Settings

## What does this PR do?

By adding a `metadata` object to the `Setting` model, we can leverage all current Plugins hooks to let plugin authors have their own configurations for plugins in the administration dashboard (you know, that place with the other `<ConfigureCard>`s)

## How do I test this PR?

Plugin «Must Read» coming soon, which has a configurable option to set a different Words Per Minute value will be coming up shortly, and should solve https://twitter.com/coralproject/status/902966149943615488

But I do need in the core in order to create the plugin (or well, I will release it with a hard-coded median WPM)
:-)

**Update 02/25/2018:** Sample plugin at https://github.com/tomasdev/talk-recipes/tree/master/plugins/admin-setting (it just adds a checkbox to enable/disable the plugin itself)
